aineko/extras/connectors/wss.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2023 Aineko Authors
+# SPDX-License-Identifier: Apache-2.0
+"""Extra module for connecting to a WebSocket."""
+
+import json
+import time
+from typing import Any, Dict, List, Optional
+
+import websocket
+from pydantic import BaseModel, field_validator
+
+from aineko import AbstractNode
+from aineko.utils.secrets import inject_secrets
+
+
+class ParamsWSS(BaseModel):
+    """Connector params for WebSocket model."""
+
+    max_retries: int = 30
+    retry_sleep: float = 5
+    url: str
+    header: Optional[Dict[str, str]] = None
+    init_messages: List[Any] = []
+    metadata: Optional[Dict[str, Any]] = None
+
+    @field_validator("url")
+    def supported_url(cls, u: str) -> str:  # pylint: disable=no-self-argument
```

Comment:
```suggestion
    @field_validator("url")
    @classmethod
    def supported_url(cls, u: str) -> str:
```
This should be decorated as a classmethod.

Reply:
fixed here: https://github.com/aineko-dev/aineko/pull/121/commits/184212cfcd0372da8112b042fe282c001ac8cc5f

note: i referenced the models directory in the codebase. maybe we should update all of these? 
e.g. https://github.com/aineko-dev/aineko/blob/develop/aineko/models/deploy_config_schema.py#L32-L37

Reply:
Good point. Let's to that in a separate PR to keep this one focused on your new feature.

Reply:
Addressed in https://github.com/aineko-dev/aineko/pull/125